nidx/nidx_relation/src/graph_query_parser.rs

```diff
@@ -0,0 +1,508 @@
+// Copyright (C) 2021 Bosutech XXI S.L.
+//
+// nucliadb is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at info@nuclia.com.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+use nidx_protos::relation_node::NodeType;
+use tantivy::collector::TopDocs;
+use tantivy::query::{AllQuery, BooleanQuery, FuzzyTermQuery, Occur, Query, TermQuery};
+use tantivy::schema::{Field, IndexRecordOption};
+use tantivy::Searcher;
+
+use crate::schema::Schema;
+use crate::{io_maps, schema};
+
+pub struct GraphSearcher {
+    schema: Schema,
+    parser: GraphQueryParser,
+    searcher: Searcher,
+}
+
+#[derive(Clone)]
+pub struct FuzzyTerm {
+    pub value: String,
+    pub fuzzy_distance: Option<u8>,
```

Comment:
Why is this an Option? Feels like it could always be specified (you have decided to do fuzzy search anyway).

Reply:
Yup, I left it from a previous design. Removing it